smarts/core/agent_manager.py

```diff
@@ -72,6 +72,7 @@ def teardown(self):
         self.teardown_ego_agents()
         self.teardown_social_agents()
         self._vehicle_with_sensors = dict()
+        self._pending_agent_ids = set()
```

Comment:
Interesting that this was not reset on `teardown()` before.

Reply:
I know.  I was surprised we never hit bugs related to this one!